addon/components/hot-replacement-component.js

```diff
@@ -61,8 +62,10 @@ const HotReplacementComponent = Ember.Component.extend(HotComponentMixin, {
   }).volatile(),
 
   __willLiveReload (event) {
-    if (matchingComponent(this.get('baseComponentName'), event.modulePath)) {
+    const baseComponentName = this.get('baseComponentName');
+    if (matchingComponent(baseComponentName, event.modulePath)) {
       event.cancel = true;
+      clearRequirejs(this, baseComponentName);
```

Comment:
@MiguelMadero This needs to be fired *before* the main `app.js` is reloaded to ensure the requirejs cache is cleared, the `clear-container-cache` is used during the actual runtime of the application to clear the cache *after* the `app.js` has been reloaded. I'm not sure what side effects moving `clear-container-cache` to *before* reloading would have, though theoretically it shouldn't have any.